packages/core/src/internal/execution/create2-execution-strategy.ts

```diff
@@ -0,0 +1,342 @@
+  534352: CREATE_X_ADDRESS,
+  3441005: CREATE_X_ADDRESS,
+  7777777: CREATE_X_ADDRESS,
+  11155111: CREATE_X_ADDRESS,
+  11155420: CREATE_X_ADDRESS,
+  68840142: CREATE_X_ADDRESS,
+  1313161554: CREATE_X_ADDRESS,
+  1313161555: CREATE_X_ADDRESS,
+  1666600000: CREATE_X_ADDRESS,
+  1666700000: CREATE_X_ADDRESS,
+};
+
+/**
+ * The most basic execution strategy, which sends a single transaction
+ * for each deployment, call, and send data, and a single static call
+ * per static call execution.
+ */
+export class Create2ExecutionStrategy implements ExecutionStrategy {
+  public readonly name: string = "create2";
+  private _createXAddress: string = CREATE_X_ADDRESS;
+
+  constructor(
+    private readonly _provider: EIP1193Provider,
+    private readonly _loadArtifact: LoadArtifactFunction
+  ) {}
+
+  public async init(): Promise<void> {
+    const client = new EIP1193JsonRpcClient(this._provider);
+    const chainId = await client.getChainId();
+
+    const existingDeployedAddress = _existingDeployedAddresses[chainId];
+
+    // if there is an existing deployment, nothing more needs done
+    if (existingDeployedAddress !== undefined) {
+      this._createXAddress = existingDeployedAddress;
+
+      return;
+    } else if (chainId === 31337) {
+      // No createX factory found because we're on a local hardhat node
+      // deploy one using presigned tx from CreateX
+      await this._provider.request({
+        method: "hardhat_setBalance",
+        params: [
+          "0xeD456e05CaAb11d66C4c797dD6c1D6f9A7F352b5",
+          "0x58D15E176280000",
+        ],
+      });
+
+      const txHash = (await this._provider.request({
+        method: "eth_sendRawTransaction",
+        params: [presignedTx],
+      })) as string;
+
+      assertIgnitionInvariant(txHash !== "0x", "CreateX deployment failed");
+
+      while (true) {
+        const receipt = await client.getTransactionReceipt(txHash);
+
+        if (receipt !== undefined) {
+          assertIgnitionInvariant(
+            receipt?.contractAddress !== undefined,
+            "CreateX deployment should have an address"
+          );
+
+          this._createXAddress = receipt.contractAddress;
+          return;
+        }
+
+        await new Promise((res) => setTimeout(res, 200));
+      }
+    } else {
+      // No createX factory found, but we're not on a local chain
+      // check if someone else has deployed CreateX on this chain
+      const result = await this._provider.request({
+        method: "eth_getCode",
+        params: [CREATE_X_ADDRESS],
+      });
+
+      assertIgnitionInvariant(
+        result !== "0x",
+        "CreateX not deployed on current network"
+      );
+
+      this._createXAddress = CREATE_X_ADDRESS;
+
+      return;
+    }
+  }
+
+  public async *executeDeployment(
+    executionState: DeploymentExecutionState
+  ): DeploymentStrategyGenerator {
+    const artifact = await this._loadArtifact(executionState.artifactId);
+    const salt = ethers.id("test");
```

Comment:
Was this intentionally left as `test` ?

Reply:
It hasn't been intentionally left as `test`. We need to review how we approach setting a salt and make sure we are happy with it.